benefits/enrollment/views.py

```diff
@@ -84,25 +87,81 @@ def index(request):
         client.oauth.ensure_active_token(client.token)
 
         funding_source = client.get_funding_source_by_token(card_token)
+        funding_source_id = funding_source.id
+        group_id = eligibility.group_id
 
         try:
-            client.link_concession_group_funding_source(funding_source_id=funding_source.id, group_id=eligibility.group_id)
+            group_funding_source = _get_group_funding_source(
+                client=client, group_id=group_id, funding_source_id=funding_source_id
+            )
+
+            already_enrolled = group_funding_source is not None
+            has_no_expiration_date = already_enrolled and group_funding_source.concession_expiry is None
+            has_expiration_date = already_enrolled and group_funding_source.concession_expiry is not None
```

Comment:
I think these `has_*` bools are a bit confusing in that they combine the notion of being enrolled with having/not having an expiration date.

When looking at their usage below, it becomes less clear that we're actually branching based on 2 pieces of data. I suggest moving the `already_enrolled and` clause down into the usage below.

Reply:
Yeah, I see what you mean. Will do

Reply:
Refactored the `if` conditions in https://github.com/cal-itp/benefits/pull/1989/commits/15bf0c9bb93fbc0a8e80a905c8422f5efbaaeb98